### PR TITLE
add command line to migrate storage

### PR DESCRIPTION
**Pull request by asf-algoo**
Experimental feature. Not for 3.8 ASF.
related to #4087

## Checkpoints

**For developers**

- [X] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [X] ~~Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)~~ Complicated and doesn't make much sense to automate.
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [X] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

## Implemented solution:

- Add `tracimcli db migrate-storage` command that permit to switch between old and new config file that have different storage name and config.

### CODE REVIEW

File: backend/tracim_backend/command/database.py [hunk 1]
```diff
@@ -296,3 +301,154 @@ def take_action(self, parsed_args: argparse.Namespace) -> None:
...
+            "-f",
+            "--force-reupload",
+            help="force reupload all existing content to new storage",
+            dest="force",
+            default=False,
+            action="store_true",
+        )
+        return parser
+
+    def _reupload_depot_file(
```
Comment (grignards) [inline backend/tracim_backend/command/database.py:331]: Wouldn't it be simpler if this function "only" converted an existing depot field to the new storage name?
Something like:
```python
def _get_new_storage_field(self, field_value, dest_storage, origin_storage, force_reupload):
   ...
  return FileIntent(file, field_value.filename, field_value.content_type),
```
Then use it like this:
```python
            for revision in revisions:
                logger.debug(
                    self, "migrate revision {} data to new storage".format(revision.revision_id)
                )
                revision.depot_file = self._get_new_storage_field(revision.depot_file, new_storage_name, original_storage_name, force_reupload)
            users = dbsession.query(User).all()
            for user in users:
                logger.debug(self, "migrate user {} data to new storage".format(user.user_id))
                user.avatar = self._get_new_storage_field(user.avatar, new_storage_name, original_storage_name, force_reupload)
                user.cropped_avatar = self._get_new_storage_field(user.cropped_avatar, new_storage_name, original_storage_name, force_reupload)
                user.cover = self._get_new_storage_field(user.cover, new_storage_name, original_storage_name, force_reupload)
                user.cropped_cover = self._get_new_storage_field(user.cropped_cover, new_storage_name, original_storage_name, force_reupload)

**Review by grignards — CHANGES_REQUESTED** [review]

File: backend/tracim_backend/command/database.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/command/database.py:331]: it do like the idea.

File: backend/tracim_backend/command/database.py [hunk 2]
```diff
@@ -296,3 +301,154 @@ def take_action(self, parsed_args: argparse.Namespace) -> None:
...
+        old_settings = get_appsettings(old_config_uri)
+        old_settings.update(old_settings.global_conf)
+        old_cfg = CFG(old_settings)
+        old_cfg.configure_filedepot()
+
+        original_storage_name = old_cfg.UPLOADED_FILES__STORAGE__STORAGE_NAME
+        original_storage_type = old_cfg.UPLOADED_FILES__STORAGE__STORAGE_TYPE
+        new_storage_name = new_cfg.UPLOADED_FILES__STORAGE__STORAGE_NAME
+        new_storage_type = new_cfg.UPLOADED_FILES__STORAGE__STORAGE_TYPE
+        print(
```
[the remark this answers is not in the document]
Comment (inkhey) [reply backend/tracim_backend/command/database.py]: done.

File: backend/tracim_backend/command/database.py (hunk 1 above)
Comment (inkhey) [reply backend/tracim_backend/command/database.py]: done.

**Review by grignards — APPROVED** [review]

**Comment by grignards** [comment]
Tested on branch.

**Pull request merged**